NXP-26872: ZIP CSV Improvements - Align with CSV Importer

eikkaj opened the pull request:
https://jira.nuxeo.com/browse/NXP-26872

nuxeojenkins commented:
[View issue in JIRA](https://jira.nuxeo.com/browse/NXP-26872)

eikkaj commented:
Created a new PR because I re-named branch according to Nuxeo guidelines. PR includes previously suggested changes.

NourNuxeo commented on nuxeo-services/nuxeo-platform-filemanager-core/src/main/java/org/nuxeo/ecm/platform/filemanager/service/extension/CSVZipImporter.java:
```diff
@@ -27,9 +27,13 @@
 import java.io.Serializable;
 import java.text.ParseException;
 import java.text.SimpleDateFormat;
+import java.util.ArrayList;
```
This and `import java.util.List;` are note used anymore since you now use streams.

NourNuxeo commented on nuxeo-services/nuxeo-platform-filemanager-core/src/test/java/org/nuxeo/ecm/platform/filemanager/TestCSVZipImporter.java:
```diff
@@ -0,0 +1,197 @@
+        ZipFile archive = CSVZipImporter.getArchiveFileIfValid(getArchiveFile("test-data/testCSVArchive.zip"));
+        assertNotNull(archive);
+        archive.close();
+    }
```
The method getArchiveFileIfValid returns null in several cases you need to try as well to make sure it behaves the way you want when.

NourNuxeo commented on nuxeo-services/nuxeo-platform-filemanager-core/src/test/java/org/nuxeo/ecm/platform/filemanager/TestCSVZipImporter.java:
```diff
@@ -0,0 +1,197 @@
+                                                         .build();
+        fm.createOrUpdateDocument(context);
+        DocumentModelList children = coreSession.getChildren(workspace1.getRef());
+        assertSame(2, children.size());
```
you shoud assertEquals on this. assertSame is to check reference equality. So the arguments are boxed without a reason.

eikkaj replied:
updated to assertEquals

NourNuxeo commented on nuxeo-services/nuxeo-platform-filemanager-core/src/test/java/org/nuxeo/ecm/platform/filemanager/TestCSVZipImporter.java:
```diff
@@ -0,0 +1,197 @@
+        assertEquals("MyNote", MyNote.getName());
+
+        assertEquals("this is text", MyNote.getProperty("note", "note"));
+        assertNotNull(MyFile.getProperty("file", "content"));
```
```suggestion
        assertEquals("this is text", MyNote.getPropertyValue("note:note"));
        assertNotNull(MyFile.getPropertyValue("file:content"));
```

NourNuxeo commented on nuxeo-services/nuxeo-platform-filemanager-core/src/test/java/org/nuxeo/ecm/platform/filemanager/TestCSVZipImporter.java:
```diff
@@ -0,0 +1,197 @@
+    public void createTestDocuments() throws Exception {
+        wsRoot = coreSession.getDocument(new PathRef("/default-domain/workspaces"));
+
+        workspace1= coreSession.createDocumentModel(wsRoot.getPathAsString(), "ws1", "Workspace");
```
Since the file is new, you can format the whole content.

NourNuxeo commented on nuxeo-services/nuxeo-platform-filemanager-core/src/test/java/org/nuxeo/ecm/platform/filemanager/TestCSVZipImporter.java:
```diff
@@ -0,0 +1,197 @@
+                                                         .build();
+        fm.createOrUpdateDocument(context);
+        DocumentModelList children = coreSession.getChildren(workspace2.getRef());
+        assertSame(0, children.size());
```
assertEquals everytime you compare two numbers that fit in a long

eikkaj replied:
updated to assertEquals

NourNuxeo commented on nuxeo-services/nuxeo-platform-filemanager-core/src/test/java/org/nuxeo/ecm/platform/filemanager/TestCSVZipImporter.java:
```diff
@@ -0,0 +1,197 @@
+            }
+         }
+    }
+}
```
missing new line at the end of file

eikkaj replied:
added line

eikkaj commented on nuxeo-services/nuxeo-platform-filemanager-core/src/main/java/org/nuxeo/ecm/platform/filemanager/service/extension/CSVZipImporter.java:
```diff
@@ -116,29 +124,32 @@ public DocumentModel createOrUpdate(FileImporterContext context) throws IOExcept
-                        } else if ("id".equalsIgnoreCase(headerValue)) {
-                            id = lineValue;
+                        } else if ("name".equalsIgnoreCase(headerValue)) {
+                            name = lineValue;
```
I'm not sure this is cleaning because it actually changes functionality, no?

NourNuxeo replied:
Yes checking the ticket which I didn't do, says you are right

eikkaj commented on nuxeo-services/nuxeo-platform-filemanager-core/src/main/java/org/nuxeo/ecm/platform/filemanager/service/extension/CSVZipImporter.java:
```diff
@@ -116,29 +124,32 @@ public DocumentModel createOrUpdate(FileImporterContext context) throws IOExcept
                         }
                     }
 
+                    if (StringUtils.isNullOrEmpty(name)) {
```
sorry, reupdated

eikkaj commented on nuxeo-services/nuxeo-platform-filemanager-core/src/main/java/org/nuxeo/ecm/platform/filemanager/service/extension/CSVZipImporter.java:
```diff
@@ -116,29 +124,32 @@ public DocumentModel createOrUpdate(FileImporterContext context) throws IOExcept
                     }
 
+                    if (StringUtils.isNullOrEmpty(name)) {
+                        log.error("Can not create or update doc without a name, skipping line");
```
logging name

eikkaj commented on nuxeo-services/nuxeo-platform-filemanager-core/src/main/java/org/nuxeo/ecm/platform/filemanager/service/extension/CSVZipImporter.java:
```diff
@@ -147,11 +158,7 @@ public DocumentModel createOrUpdate(FileImporterContext context) throws IOExcept
                             log.error("Can not create doc without a type, skipping line");
```
logging name

efge commented on nuxeo-services/nuxeo-platform-filemanager-core/src/main/java/org/nuxeo/ecm/platform/filemanager/service/extension/CSVZipImporter.java:
```diff
@@ -59,13 +63,16 @@
 import org.nuxeo.ecm.platform.filemanager.api.FileImporterContext;
 
+import static org.nuxeo.ecm.core.api.LifeCycleConstants.INITIAL_LIFECYCLE_STATE_OPTION_NAME;
+import static org.nuxeo.ecm.core.query.sql.NXQL.ECM_LIFECYCLESTATE;
```
It's a detail but our standard formatter puts static imports before regular ones.

eikkaj replied:
updated

eikkaj commented:
I added a second Cleanup / Format commit but left it as the second commit because I renamed/moved the TestZIPCSV class from TestCSV so it must be cleaned up after the fact.

eikkaj commented:
I have added a new commit where I've combined the test archive files to simplify testing
